util/py/packages/lib/logging_.py

```diff
@@ -0,0 +1,28 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+from enum import Enum
+
+from rich.logging import RichHandler
+
+from util.py.packages.lib.wrapper import Wrapper
+
+log = Wrapper()
+
+
+class LogLevel(str, Enum):
+    ERROR: str = "error"
+    WARNING: str = "warning"
+    INFO: str = "info"
+    DEBUG: str = "debug"
+
+
+def init(log_level: LogLevel = LogLevel.WARNING):
+    global log
+    logging.basicConfig(level=log_level.upper(),
+                        format="%(message)s",
+                        datefmt="[%X]",
+                        handlers=[RichHandler()])
+    log.replace_wrapped(logging.getLogger("rich"))
```

Comment:
Yes, it does. From the docstring of `Wrapper`:
>A thin wrapper that allows replacing an already referenced object.                                                                                                                                                                                                   
>This allows updating a global after it has been imported by other modules.

Without this, other modules would not be able to just import `log` since it's initialized after it is imported. The assignment would replace the `log` in `logging_.py` but other modules that already imported it would still see `None` (or whatever the initial value of `log` was). We could define a function that returns the `log` from `logging_.py` but `Wrapper` solves this in a more reusable way.